spirv_msl.cpp

```diff
@@ -393,6 +428,15 @@ uint32_t CompilerMSL::add_interface_block(StorageClass storage)
 	MemberSorter memberSorter(ib_type, meta[ib_type_id], sort_aspect);
 	memberSorter.sort();
 
+	// Sort input or output variables alphabetical
+	auto &execution = get_entry_point();
+	if ((execution.model == ExecutionModelFragment && storage == StorageClassInput) ||
+	    (execution.model == ExecutionModelVertex && storage == StorageClassOutput))
+	{
+		MemberSorter memberSorter(ib_type, meta[ib_type.self], MemberSorter::Alphabetical);
+		memberSorter.sort();
+	}
```

Comment:
I haven't looked at this in detail, so I might be repeating stuff, but if a shader does declare DecorationLocation on varyings/attributes, this should always be honored.

I guess the question here is what the MSL backend should do in case that there is no information in the shader itself?